### ISSUE TITLE
Extend patternMaker class

### ISSUE THREAD

**Issue by dcallus** [fillmaster/fillMaster#42]
assigned to [ninad](https://github.com/ninad-d)

patternMaker class needs a parent class where most of the logic will live.
We also need a new child class.

- [ ] Create parent class for patternMaker called patternMakerBase
- [ ] Remove logic relating to countIn (from patternMakerBase) so it's only present in patternMaker
- [ ] Remove logic relating to fills (from patternMakerBase) so it's only present in patternMaker
- [ ] Remove singleton logic (so that multiple instances are possible) from patternMakerBase, but keep it in patternMaker, if possible. Let me know if it's not. 
- [ ] Make a new child (of patternMakerBase) called TempoPreviewMaker.
The main issue:
- [ ] Change the logic somehow so that the numbers used in the metronome strings (i.e. 1, 2, 3) are less fixed. Probably start by putting the defaults in the constructor instead of at the top of the class. Then do something else with the logic. Not sure what yet. Please see if you can come up with something.  For example, the implementation of createMetronomeString includes a flag for isCountIn. I think the isCountIn should be separated and its own method. The result should be the same, an output string of only 1s and 0s. The TempoPreviewMaker should only output 1s and 0s when calling createMetronomeString, but of course it's currently hardcoded to output 1s 2s and 0s. Hope this makes sense. If not just ask

**Comment by ninad-d** [comment]
@dcallus on it.

**Comment by ninad-d** [comment]
@dcallus if you have merged the pattern maker base changes then I will move to remaining items in the list. I guess we are able to address the first four points with current changes.

**Comment by dcallus** [comment]
> @dcallus if you have merged the pattern maker base changes then I will move to remaining items in the list. I guess we are able to address the first four points with current changes.

Think I edited your message instead of quotes. sorry.

Hey sorry I was having some issues. New Pattern and metronomePattern classes are now fully working, tests are passing again.

I copied your logic for the patternSettings and applied it to the playHelperOn and changed everywhere that sets these things.. I've merged all the changes into the epic branch. Just going to rewrite the tasks there as they are quite unclear and will try and separate out the UI stuff, I'll try focus on that more myself.
